modules/certificate/certificate.service.js

```diff
@@ -42,6 +42,22 @@ class CertificatesService {
     return certificate;
   }
 
+  async generateCertificate(value, id) {
+    let candidate;
+    let name;
+    let firstNamePart;
+    let secondNamePart;
+
+    do {
```

Comment:
I think it is better to use recursion for certificate generation.

Reply:
Ok, I will fix it